rust/private/rustc.bzl

```diff
@@ -965,6 +965,10 @@ def construct_arguments(
     if linker_script:
         rustc_flags.add(linker_script, format = "--codegen=link-arg=-T%s")
 
+    if hasattr(ctx.attr, "_experimental_toolchain_generated_sysroot"):
```

Comment:
Can this be safely added unconditionally? Or are there rules that will break when this check is removed?

Reply:
Good point. Some rules don't use `common_attrs` but we should add `_experimental_toolchain_generated_sysroot` to all rules that use `rustc`.

Reply:
Which rules break?

Reply:
`rustdoc`.

Reply:
`rustdoc` directly calls `construct_arguments`, maybe restoring the parameter to this function would do the trick?

Reply:
I think we should just add the attribute to `rustdoc` rule. https://github.com/bazelbuild/rules_rust/pull/2223/commits/b1a7e1a318faddaff0a486bf36e68c9af313c4b2#diff-ac2237497d8db708da38cbb6a5fda472b241e592c0c148c215eb7687c11b4d1f

Reply:
Oh, I thought there were some rules that didn't support `--sysroot`. If that's not the case then this approach seems good to me!

Reply:
> Oh, I thought there were some rules that didn't support `--sysroot`. If that's not the case then this approach seems good to me!

Ok. Sounds good to me.